### PR TITLE
Take the latest drop of CsWin32

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Take the latest drop of CsWin32.

- OLE_HANDLE is now defined
- VARIANT_BOOL is now defined
- COM structs now implement a stable Guid and interface
- Some COM getters and setters have turned into properties

Cleaned up a few suggestion messages in affected files.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8225)

### CODE REVIEW

File: src/System.Windows.Forms.Primitives/src/Windows/Win32/System/Ole/CADWORD.cs [hunk 1]
```diff
@@ -6,6 +6,17 @@
 
 namespace Windows.Win32.System.Ole;
 
+internal partial struct OLE_HANDLE
```
Comment (lonitra) [inline src/System.Windows.Forms.Primitives/src/Windows/Win32/System/Ole/CADWORD.cs:9]: Should this be in its own file?

File: src/System.Windows.Forms.Primitives/src/System/Windows/Forms/AccessibiltyExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+using Accessibility;
+
+namespace System.Windows.Forms
+{
+    internal static unsafe class AccessibiltyExtensions
+    {
+        /// <inheritdoc cref="PInvoke.LresultFromObject(Guid*, WPARAM, IUnknown*)"/>
+        internal static LRESULT GetLRESULT(this IAccessible accessible, WPARAM wparam)
+        {
+            // https://docs.microsoft.com/windows/win32/winauto/how-to-handle-wm-getobject
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Primitives/src/System/Windows/Forms/AccessibiltyExtensions.cs:15]: the new link is https://learn.microsoft.com/windows/win32/winauto/how-to-handle-wm-getobject

File: src/System.Windows.Forms.Primitives/src/Windows/Win32/System/Ole/CADWORD.cs (hunk 1 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms.Primitives/src/Windows/Win32/System/Ole/CADWORD.cs:9]: Whoopsie

File: src/System.Windows.Forms.Primitives/src/System/Windows/Forms/AccessibiltyExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+            using var unknown = ComHelpers.GetComScope<IUnknown>(accessible, out HRESULT hr);
```
[the remark this answers is not in the document]
Comment (JeremyKuhne) [reply src/System.Windows.Forms.Primitives/src/System/Windows/Forms/AccessibiltyExtensions.cs:17]: I'm hesitant as I think it is expected to fail- concerned about noise

File: src/System.Windows.Forms.Primitives/src/Windows/Win32/Foundation/IID.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace Windows.Win32.Foundation
+{
+    internal static unsafe class IID
+    {
+        // We cast away the "readonly" here as there is no way to communicate that through a pointer and
+        // Marshal APIs take the Guid as ref. Even though none of our usages actually change the state.
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Guid* Get<T>() where T : unmanaged, IComIID
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Primitives/src/Windows/Win32/Foundation/IID.cs:15]: Would it make sense to name this methos GetPointer?

**Review by Tanya-Solyanik — APPROVED** [review]

File: src/System.Windows.Forms/src/System/Windows/Forms/AccessibleObject.cs [hunk 5]
```diff
@@ -1801,14 +1800,13 @@ protected void UseStdAccessibleObjects(IntPtr handle, int objid)
             UnsafeNativeMethods.CreateStdAccessibleObject(
                 new HandleRef(this, handle),
                 objid,
-                ref IID.IAccessible,
+                ref IID.GetRef<global::Windows.Win32.UI.Accessibility.IAccessible>(),
```
[the remark this answers is not in the document]
Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/AccessibleObject.cs:1803]: Can't for `IAccessible` because it's public. :(

**Pull request merged**